igvc_navigation/src/trajectory_planner/motion_profiler.cpp

```diff
@@ -0,0 +1,86 @@
+#include "motion_profiler.h"
+#include <tf/transform_datatypes.h>
+
+MotionProfiler::MotionProfiler(double axle_length, const WheelConstraint& wheel_constraint,
+                               const RobotConstraint& robot_constraint,
+                               const MotionProfilerOptions& motion_profiler_options, double target_velocity)
+  : wheel_constraint_{ wheel_constraint }
+  , robot_constraint_{ robot_constraint }
+  , motion_profiler_options_{ motion_profiler_options }
+  , axle_length_{ axle_length }
+  , target_velocity_{ target_velocity }
+{
+}
+
+void MotionProfiler::profileTrajectory(const igvc_msgs::trajectoryPtr& trajectory_ptr, const RobotState& state)
```

Comment:
With how you currently are running things, you really don't gain anything by iterating over the entire trajectory. You only need to put a velocity profile over the segment between the current and next trajectory points, unless you're doing the backward pass (you're not).

Reply:
the original idea here was that it would be easier to implement something that did need to iterate over the entire trajectory in the future when we decide to make this better, but yeah you're right that iterating over the entire trajectory doesn't do anything here.